### PR TITLE
N8N-3750 Redesign button component

### PR DESCRIPTION
**Pull request by alexgrozav**

### CODE REVIEW

File: packages/design-system/src/components/N8nIconButton/IconButton.vue [hunk 1]
```diff
@@ -23,13 +13,15 @@ export default {
...
-			type: String,
+			default: 'primary',
+			validator: (value: string): boolean =>
+				['primary', 'secondary', 'tertiary', 'success', 'warning', 'danger'].includes(value),
 		},
 		size: {
 			type: String,
 			default: 'medium',
+			validator: (value: string): boolean =>
+				['mini', 'small', 'medium', 'large', 'xlarge'].includes(value),
```
Comment (mutdmour) [inline packages/design-system/src/components/N8nIconButton/IconButton.vue:24]: would recommend removing validation here.. that way `n8n-button` component can handle validation. and we will get one warning instead of 2 warnings.. also don't have to update this one if we change that

File: packages/design-system/theme/src/_tokens.scss [hunk 2]
```diff
@@ -1,405 +1,448 @@
...
-	--spacing-l: 1.5rem;
-	--spacing-xl: 2rem;
-	--spacing-2xl: 3rem;
-	--spacing-3xl: 4rem;
-	--spacing-4xl: 8rem;
-	--spacing-5xl: 16rem;
+  --color-primary-h: 7;
+  --color-primary-s: 100%;
+  --color-primary-l: 68%;
+  --color-primary: hsl(
```
Comment (mutdmour) [inline packages/design-system/theme/src/_tokens.scss:7]: it looks like you changed a few different colors here in this file... was this intentional? can you double check with Max? Pretty sure the floats make a difference..

File: packages/editor-ui/vue.config.js [hunk 3]
```diff
@@ -27,6 +28,12 @@ module.exports = {
 		plugins: [
 			new MonacoWebpackPlugin({ languages: ['javascript', 'json', 'typescript'] }),
 		],
+		resolve: {
+			alias: {
+				'element-ui/packages/button': path.resolve(__dirname, '..', 'design-system/src/components/N8nButton/Button.vue'),
+				'element-ui/lib/button': path.resolve(__dirname, '..', 'design-system/src/components/N8nButton/Button.vue'),
```
Comment (mutdmour) [inline packages/editor-ui/vue.config.js:34]: nice, though I expected this to link to a wrapper component... that maps element ui props to our own props... that way if they define size `mini` we can translate that to our size `2xsmall` for example.. also won't break how it's used by other element ui elements. though button might be too simple for this..

File: packages/editor-ui/src/components/NodeExecuteButton.vue [hunk 4]
```diff
@@ -4,6 +4,7 @@
 		:disabled="workflowRunning && !nodeRunning"
 		:label="buttonLabel"
 		:type="type"
+		:outline="outline"
 		:size="size"
 		:transparentBackground="transparent"
```
Comment (mutdmour) [inline packages/editor-ui/src/components/NodeExecuteButton.vue:9]: guess this needs to be deleted

File: packages/editor-ui/src/components/FixedCollectionParameter.vue [hunk 5]
```diff
@@ -287,6 +287,13 @@ export default mixins(genericHelpers)
 </script>
 
 <style scoped lang="scss">
+::v-deep {
+	.button {
+		--button-background-color: var(--color-background-base);
+		--button-border-color: var(--color-foreground-base);
+	}
+}
```
Comment (mutdmour) [inline packages/editor-ui/src/components/FixedCollectionParameter.vue:295]: we should try to avoid adding overrides as much as possible. why are these needed when it was not needed before? can we not add a variation of this theme that works? (applies to the two other overrides in this PR)
if current impl requires this, talk to Max, he can recommend a way to use button without overrides.

File: packages/design-system/src/components/N8nIconButton/IconButton.vue (hunk 1 above)
Comment (alexgrozav) [reply packages/design-system/src/components/N8nIconButton/IconButton.vue]: Good suggestion! Removed.

File: packages/design-system/theme/src/_tokens.scss (hunk 2 above)
Comment (alexgrozav) [reply packages/design-system/theme/src/_tokens.scss]: It resolves to the same hex code, but reverted the changes. Thanks!

File: packages/editor-ui/src/components/FixedCollectionParameter.vue (hunk 5 above)
Comment (alexgrozav) [reply packages/editor-ui/src/components/FixedCollectionParameter.vue:295]: Unfortunately, we've got a special case here where the tertiary button must have a different background. I've discussed it with Max and there's a planned replacement for the way we visualize the fields in NDV and we're ok to compromise for now.

File: packages/design-system/theme/src/_tokens.scss [hunk 6]
```diff
@@ -1,405 +1,448 @@
...
+  --color-primary-tint-1-l: 88%;
+  --color-primary-tint-1: hsl(
+				  var(--color-primary-tint-2-h),
+				  var(--color-primary-tint-2-s),
+				  var(--color-primary-tint-1-l)
+  );
+
+  --color-primary-tint-2-h: 11;
+  --color-primary-tint-2-s: 100%;
+  --color-primary-tint-2-l: 95%;
```
Comment (mutdmour) [inline packages/design-system/theme/src/_tokens.scss:24]: can you make sure to revert all values that changed?

Comment (alexgrozav) [reply packages/design-system/theme/src/_tokens.scss]: Yes, sorry. I updated the values to reflect what's in Figma at first, but spoke to Max about it afterwards and found out they're not a 100% match to make theming easier. Updated.

**Review by mutdmour — APPROVED** [review]

**Comment by janober** [comment]
Got released with `n8n@0.187.1`
